extensions/objects/templates/TranslatableHtmlEditorDirective.js

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a translatable html editor.
+ */
+
+oppia.directive('translatableHtmlEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getInitArgs: '&',
+        value: '='
+      },
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/objects/templates/translatable_html_editor_directive.html'),
+      controller: ['$scope', function($scope) {
+        if ($scope.value.content_id === '') {
+          $scope.value.content_id = 'interaction_' + (
+            Math.random().toString(36).slice(-5));
```

Comment:
Ah. I am a bit dubious about this.

Can we pre-populate the content ID in advance? I don't think this directive should be in charge of doing that. It's just supposed to display the editor; assigning a content ID is out of its scope.